src/components/Launcher.js

```diff
@@ -10,33 +10,51 @@ class Launcher extends Component {
   constructor() {
     super();
     this.state = {
-      active: false,
-      launcherIcon
+      launcherIcon,
+      isOpen: false
     };
   }
 
+  displayNewMessagesCount() {
+    if (this.props.newMessagesCount !== 0 && this.state.isOpen === false) {
+      return (
+        <div className={"sc-new-messsages-count"}>
+          {this.props.newMessagesCount}
+        </div>
+      )
+    } 
+  }
+
   handleClick() {
-    this.setState({
-      active: !this.state.active
-    });
+    if (this.props.handleClick !== undefined) {
+      this.props.handleClick();
+    } else {
+      this.setState({
+        isOpen: !this.state.isOpen,
+      });
+    }
   }
 
   render() {
+    const isOpen = (this.props.isOpen === undefined) ? this.state.isOpen : this.props.isOpen;
     const classList = [
       'sc-launcher',
-      (this.state.active ? ' active' : ''),
+      (isOpen ? ' opened' : ''),
     ];
     return (
       <div>
+        <div>
+        </div>
         <div className={classList.join(' ')} onClick={this.handleClick.bind(this)}>
+          {this.displayNewMessagesCount()}
```

Comment:
I think the name `displayNewMessagesCount` makes it seem like this is imperative, like this function when called actually does something to "display" the message count. But it doesn't all it does is conditionally return a div.

I think a more imperative, react-ish way to do this would be to use a pure-functional component for the message count:

``` jsx
const MessageCount = (props) => {
     // some conditions or whatever
      return <div...
}
```

``` jsx
{this.state.isOpen && <MessageCount count={something}/>}
```